server/src/main/java/org/opensearch/common/Stream.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common;
```

Comment:
Move to org.opensearch.common.io. We already have a bunch a Stream utility in that package, do we think we can leverage few pieces?

Reply:
+1 to the holder part and naming of class part of comment. For content length, since we need an InputStream interface here to allow this to contain different types of streams and at the same time associate length of each of them, there is no way to get it from InputStream. Because of the same reason to avoid having inconsistent behavior for a stream to not have content length equal to passed length, we are packing the two together. Content length here specifies how much content we want to read from the stream.